module/zfs/spa.c

```diff
@@ -2197,16 +2198,16 @@ spa_load_verify_done(zio_t *zio)
 	}
 
 	mutex_enter(&spa->spa_scrub_lock);
-	spa->spa_load_verify_ios--;
+	spa->spa_load_verify_bytes -= BP_GET_PSIZE(bp);
 	cv_broadcast(&spa->spa_scrub_io_cv);
 	mutex_exit(&spa->spa_scrub_lock);
 }
 
 /*
- * Maximum number of concurrent scrub i/os to create while verifying
- * a pool while importing it.
+ * Maximum number of inflight bytes is the log2 faction of the arc size.
+ * By default, we set it to 1/16th of the arc.
  */
-int spa_load_verify_maxinflight = 10000;
+int spa_load_verify_shift = 4;
```

Comment:
Why is this tunable expressed as ratio of the ARC?
Can you mention the reason for that decision in the comment?

Reply:
The ARC is already some portion of all memory and as we do the traversal we will end up caching blocks in the ARC. So it made sense to base the max number of inflight bytes on how much of the ARC they will consume. This is also a common calculation that is used by other subsystems within ZFS and overall a conservative approach to ensure we don't overload the system.